Panic due to unexpected Nil when reading custom webhook resources during plan action

**Describe the bug**
When running a plan on deployed custom webhook resources, the `pkg/artifactory/resource/webhook/resource_artifactory_webhook_release_bundle.go` panics due to an unexpected nil value. The resource is not used/present in the current state.

**Requirements for and issue**
- [x] A description of the bug
- [x] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue). **If this is not supplied, this issue will likely be closed without any effort expended.**
- [x] Your version of artifactory: SaaS, latest
- [x] Your version of terraform: `Terraform v1.9.8`
- [x] Your version of terraform provider: `v12.8.3`

**Expected behavior**
I expect the Read operation to run successfully

**Additional context**
The github token value is an empty string, so the custom webhooks currently run into 401 errors when run. 

What I find especially strange is that I am not explicitly using the resource `artifactory_release_bundle_custom_webhook`. 

My state looks like so:

```shell
module.event_routing["rnd"].artifactory_artifact_custom_webhook.this
module.event_routing["rnd"].artifactory_artifact_property_custom_webhook.this
module.event_routing["rnd"].artifactory_build_custom_webhook.this
module.event_routing["rnd"].artifactory_destination_custom_webhook.this
module.event_routing["rnd"].artifactory_docker_custom_webhook.this
module.event_routing["rnd"].artifactory_user_custom_webhook.this
```

Configs:

<details>

<summary>Module caller</summary>

```hcl
module "event_routing" {
  /*
    Configures Custom Webhooks for all available Artifactory events emitted by all projects and repositories to
    be forwarded to the selcted GHE organization's Self-Serivce repository via the GitHub `repository-dispatch` API.
    */
  for_each = toset([
    "rnd",
  ])
  source = "./webhook-module"

  target_org   = each.value
  github_token = var.github_token
}

variable "github_token" {
  description = "The GitHub token to use for dispatching events"
  type        = string
  sensitive   = true
  default     = ""
}
```

</details>

<details>

<summary>Module code: </summary> 

```hcl
variable "target_org" {
  description = "The organization to which the webhook events should be forwarded to"
  type        = string
}

variable "repo_keys" {
  description = "The repository keys to which the webhook events should be forwarded to"
  type        = list(string)
  default     = []
}

variable "github_token" {
  description = "The GitHub token to use for the webhook"
  type        = string
  sensitive   = true
}

terraform {
  required_providers {
    artifactory = {
      source  = "jfrog/artifactory"
      version = "~> 12.0"
    }
  }
}

resource "artifactory_artifact_custom_webhook" "this" {
  key     = "${var.target_org}-artifact-events-webhook"
  enabled = true
  event_types = [
    "deployed",
    "deleted",
    "copied",
    "moved",
    "cached",
  ]

  criteria {
    any_local        = true
    any_remote       = true
    any_federated    = true
    repo_keys        = var.repo_keys
    include_patterns = ["**/*"]
  }

  handler {
    url = "https://api.github.com/repos/${var.target_org}/swae-${var.target_org}/dispatches"
    # method = "POST"  // TODO: Uncomment once provider was upgraded
    secrets = {
      token = var.github_token
    }
    http_headers = {
      "Content-Type"  = "application/json"
      "authorization" = "Bearer {{ .secrets.token }}"
    }
    payload = "{ \"event_type\": \"artifact/{{ .type }}\", \"client_payload\": {{ json .data}} }"
  }
}

resource "artifactory_artifact_lifecycle_custom_webhook" "this" {
  key     = "${var.target_org}-artifact-lifecycle-events-webhook"
  enabled = true
  event_types = [
    "archive",
    "restore",
  ]

  handler {
    url = "https://api.github.com/repos/${var.target_org}/swae-${var.target_org}/dispatches"
    # method = "POST"  // TODO: Uncomment once provider was upgraded
    secrets = {
      token = var.github_token
    }
    http_headers = {
      "Content-Type"  = "application/json"
      "authorization" = "Bearer {{ .secrets.token }}"
    }
    payload = "{ \"event_type\": \"artifact_lifecycle/{{ .type }}\", \"client_payload\": {{ json .data}} }"
  }
}

resource "artifactory_artifact_property_custom_webhook" "this" {
  key     = "${var.target_org}-artifact-property-events-webhook"
  enabled = true
  event_types = [
    "added",
    "deleted",
  ]

  criteria {
    any_local        = true
    any_remote       = true
    any_federated    = true
    repo_keys        = var.repo_keys
    include_patterns = ["**/*"]
  }

  handler {
    url = "https://api.github.com/repos/${var.target_org}/swae-${var.target_org}/dispatches"
    # method = "POST"  // TODO: Uncomment once provider was upgraded
    secrets = {
      token = var.github_token
    }
    http_headers = {
      "Content-Type"  = "application/json"
      "authorization" = "Bearer {{ .secrets.token }}"
    }
    payload = "{ \"event_type\": \"artifact_property/{{ .type }}\", \"client_payload\": {{ json .data}} }"
  }
}


resource "artifactory_build_custom_webhook" "this" {
  key     = "${var.target_org}-build-events-webhook"
  enabled = true
  event_types = [
    "uploaded",
    "deleted",
    "promoted",
  ]

  criteria {
    any_build        = true
    selected_builds  = []
    include_patterns = ["**/*"]
  }

  handler {
    url = "https://api.github.com/repos/${var.target_org}/swae-${var.target_org}/dispatches"
    # method = "POST"  // TODO: Uncomment once provider was upgraded
    secrets = {
      token = var.github_token
    }
    http_headers = {
      "Content-Type"  = "application/json"
      "authorization" = "Bearer {{ .secrets.token }}"
    }
    payload = "{ \"event_type\": \"build/{{ .type }}\", \"client_payload\": {{ json .data}} }"
  }
}


resource "artifactory_destination_custom_webhook" "this" {
  key     = "${var.target_org}-destination-events-webhook"
  enabled = true
  event_types = [
    "received",
    "delete_started",
    "delete_completed",
    "delete_failed",
  ]

  criteria {
    any_release_bundle              = true
    registered_release_bundle_names = []
    include_patterns                = ["**/*"]
  }

  handler {
    url = "https://api.github.com/repos/${var.target_org}/swae-${var.target_org}/dispatches"
    # method = "POST"  // TODO: Uncomment once provider was upgraded
    secrets = {
      token = var.github_token
    }
    http_headers = {
      "Content-Type"  = "application/json"
      "authorization" = "Bearer {{ .secrets.token }}"
    }
    payload = "{ \"event_type\": \"destination/{{ .type }}\", \"client_payload\": {{ json .data}} }"
  }
}


resource "artifactory_distribution_custom_webhook" "this" {
  key     = "${var.target_org}-distribution-events-webhook"
  enabled = true
  event_types = [
    "distribute_started",
    "distribute_completed",
    "distribute_failed",
    "distribute_aborted",
    "delete_started",
    "delete_completed",
    "delete_failed",
  ]

  criteria {
    any_release_bundle              = true
    registered_release_bundle_names = []
    include_patterns                = ["**/*"]
  }

  handler {
    url = "https://api.github.com/repos/${var.target_org}/swae-${var.target_org}/dispatches"
    # method = "POST"  // TODO: Uncomment once provider was upgraded
    secrets = {
      token = var.github_token
    }
    http_headers = {
      "Content-Type"  = "application/json"
      "authorization" = "Bearer {{ .secrets.token }}"
    }
    payload = "{ \"event_type\": \"distribution/{{ .type }}\", \"client_payload\": {{ json .data}} }"
  }
}


resource "artifactory_docker_custom_webhook" "this" {
  key     = "${var.target_org}-docker-events-webhook"
  enabled = true
  event_types = [
    "pushed",
    "deleted",
    "promoted",
  ]

  criteria {
    any_local        = true
    any_remote       = true
    any_federated    = true
    repo_keys        = var.repo_keys
    include_patterns = ["**/*"]
  }


  handler {
    url = "https://api.github.com/repos/${var.target_org}/swae-${var.target_org}/dispatches"
    # method = "POST"  // TODO: Uncomment once provider was upgraded
    secrets = {
      token = var.github_token
    }
    http_headers = {
      "Content-Type"  = "application/json"
      "authorization" = "Bearer {{ .secrets.token }}"
    }
    payload = "{ \"event_type\": \"docker/{{ .type }}\", \"client_payload\": {{ json .data}} }"
  }
}


resource "artifactory_user_custom_webhook" "this" {
  key     = "${var.target_org}-user-events-webhook"
  enabled = true
  event_types = [
    "locked",
  ]

  handler {
    url = "https://api.github.com/repos/${var.target_org}/swae-${var.target_org}/dispatches"
    # method = "POST"  // TODO: Uncomment once provider was upgraded
    secrets = {
      token = var.github_token
    }
    http_headers = {
      "Content-Type"  = "application/json"
      "authorization" = "Bearer {{ .secrets.token }}"
    }
    payload = "{ \"event_type\": \"user/{{ .type }}\", \"client_payload\": {{ json .data}} }"
  }
}

```

</details>

**Stacktrace**
```shell

Stack trace from the terraform-provider-artifactory_v12.8.3 plugin:

panic: interface conversion: interface {} is nil, not bool

goroutine 323 [running]:
github.com/jfrog/terraform-provider-artifactory/v12/pkg/artifactory/resource/webhook.fromReleaseBundleCriteriaAPIModel({0x1367308, 0xc000419890}, 0xc00097bce0, 0xc000ed00c0)
        github.com/jfrog/terraform-provider-artifactory/v12/pkg/artifactory/resource/webhook/resource_artifactory_webhook_release_bundle.go:316 +0x798
github.com/jfrog/terraform-provider-artifactory/v12/pkg/artifactory/resource/webhook.(*ReleaseBundleCustomWebhookResourceModel).fromAPIModel(0xc0003be340, {0x1367308, 0xc000419890}, {{{0xc000a00b70, 0x26}, {0x0, 0x0}, 0x1, {{0xc000d0ce10, 0xb}, ...}, ...}, ...}, ...)
        github.com/jfrog/terraform-provider-artifactory/v12/pkg/artifactory/resource/webhook/resource_artifactory_custom_webhook_release_bundle.go:235 +0x19b
github.com/jfrog/terraform-provider-artifactory/v12/pkg/artifactory/resource/webhook.(*ReleaseBundleCustomWebhookResource).Read(0xc0002f61c0, {0x1367308, 0xc000419890}, {{{{0x136e3b8, 0xc0008c4270}, {0x10109a0, 0xc0005f5710}}, {0x1370b38, 0xc000c86230}}, 0xc000a982a8, ...}, ...)
        github.com/jfrog/terraform-provider-artifactory/v12/pkg/artifactory/resource/webhook/resource_artifactory_custom_webhook_release_bundle.go:146 +0x5ae
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0xc00041a248, {0x1367308, 0xc000419890}, 0xc0004198f0, 0xc000b73508)
        github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/fwserver/server_readresource.go:117 +0x86e
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadResource(0xc00041a248, {0x1367308?, 0xc000419770?}, 0xc000968dc0)
        github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/proto6server/server_readresource.go:55 +0x38e
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ReadResource(0xc0002aa000, {0x1367308?, 0xc000419440?}, 0xc000968dc0)
        github.com/hashicorp/terraform-plugin-mux@v0.18.0/tf6muxserver/mux_server_ReadResource.go:35 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadResource(0xc0000e4140, {0x1367308?, 0xc000418b40?}, 0xc0002f6150)
        github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/tf6server/server.go:785 +0x309
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadResource_Handler({0x1145ce0, 0xc0000e4140}, {0x1367308, 0xc000418b40}, 0xc00084e800, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:575 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00049e200, {0x1367308, 0xc000418ab0}, 0xc0009dcde0, 0xc0003bb0e0, 0x1b37d28, 0x0)
        google.golang.org/grpc@v1.69.4/server.go:1392 +0xfd2
google.golang.org/grpc.(*Server).handleStream(0xc00049e200, {0x1367ba0, 0xc000400000}, 0xc0009dcde0)
        google.golang.org/grpc@v1.69.4/server.go:1802 +0xbaa
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.69.4/server.go:1030 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 65
        google.golang.org/grpc@v1.69.4/server.go:1041 +0x125

Error: The terraform-provider-artifactory_v12.8.3 plugin crashed!
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,8 +1,9 @@
-## 12.8.4 (Februry 7, 2025). Tested on Artifactory 7.104.6 with Terraform 1.10.5 and OpenTofu 1.9.0
+## 12.8.4 (Februry 12, 2025). Tested on Artifactory 7.104.6 with Terraform 1.10.5 and OpenTofu 1.9.0
 
 BUG FIXES:
 
 * resource/artifactory_\*\_repository: Fix error when provider tries to refresh the resource for repository that is no longer existed on Artifactory. Issue: [#1189](https://github.com/jfrog/terraform-provider-artifactory/issues/1189) PR: [#1190](https://github.com/jfrog/terraform-provider-artifactory/pull/1190)
+* resource/artifactory_artifactory_release_bundle_custom_webhook, resource/artifactory_destination_custom_webhook, resource/artifactory_distribution_custom_webhook, resource/artifactory_release_bundle_custom_webhook: Fix provider panic crash. Issue: [#1192](https://github.com/jfrog/terraform-provider-artifactory/issues/1192) PR: [#1196](https://github.com/jfrog/terraform-provider-artifactory/pull/1196)
 
 ## 12.8.3 (January 28, 2025). Tested on Artifactory 7.98.14 with Terraform 1.10.5 and OpenTofu 1.9.0
 

--- a/pkg/artifactory/resource/webhook/resource_artifactory_custom_webhook_release_bundle_test.go
+++ b/pkg/artifactory/resource/webhook/resource_artifactory_custom_webhook_release_bundle_test.go
@@ -1,0 +1,93 @@
+package webhook_test
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/jfrog/terraform-provider-artifactory/v12/pkg/acctest"
+	"github.com/jfrog/terraform-provider-shared/testutil"
+	"github.com/jfrog/terraform-provider-shared/util"
+)
+
+func TestAccCustomWebhook_Distribution(t *testing.T) {
+	jfrogURL := os.Getenv("JFROG_URL")
+	if !strings.HasSuffix(jfrogURL, "jfrog.io") {
+		t.Skipf("env var JFROG_URL '%s' is not a cloud instance. It also needs to have distribution enabled.", jfrogURL)
+	}
+
+	_, fqrn, name := testutil.MkNames("test-distribution-custom-webhook", "artifactory_distribution_custom_webhook")
+
+	params := map[string]interface{}{
+		"webhookName": name,
+		"payload":     `{ \"event_type\": \"distribution/{{ .type }}\", \"client_payload\": {{ json .data}} }`,
+	}
+	webhookConfig := util.ExecuteTemplate("TestAccCustomWebhookDistribution", `
+		resource "artifactory_distribution_custom_webhook" "{{ .webhookName }}" {
+			key         = "{{ .webhookName }}"
+			description = "test description"
+			enabled     = true
+			event_types = ["distribute_started", "distribute_completed", "delete_started", "delete_completed"]
+
+			criteria {
+				any_release_bundle              = true
+				registered_release_bundle_names = []
+				include_patterns                = ["**/*"]
+			}
+
+			handler {
+				url = "https://google.com"
+				method = "POST"
+				secrets = {
+					secret1 = "value1"
+					secret2 = "value2"
+				}
+				http_headers = {
+					header-1 = "value-1"
+					header-2 = "value-2"
+				}
+				payload = "{{ .payload }}"
+			}
+		}
+	`, params)
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		CheckDestroy:             acctest.VerifyDeleted(t, fqrn, "key", testCheckWebhook),
+		Steps: []resource.TestStep{
+			{
+				Config: webhookConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(fqrn, "event_types.#", "4"),
+					resource.TestCheckTypeSetElemAttr(fqrn, "event_types.*", "distribute_started"),
+					resource.TestCheckTypeSetElemAttr(fqrn, "event_types.*", "distribute_completed"),
+					resource.TestCheckTypeSetElemAttr(fqrn, "event_types.*", "delete_started"),
+					resource.TestCheckTypeSetElemAttr(fqrn, "event_types.*", "delete_completed"),
+					resource.TestCheckResourceAttr(fqrn, "criteria.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "criteria.0.any_release_bundle", "true"),
+					resource.TestCheckResourceAttr(fqrn, "criteria.0.registered_release_bundle_names.#", "0"),
+					resource.TestCheckResourceAttr(fqrn, "criteria.0.include_patterns.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "criteria.0.include_patterns.0", "**/*"),
+					resource.TestCheckResourceAttr(fqrn, "handler.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "handler.0.url", "https://google.com"),
+					resource.TestCheckResourceAttr(fqrn, "handler.0.method", "POST"),
+					resource.TestCheckResourceAttr(fqrn, "handler.0.secrets.secret1", "value1"),
+					resource.TestCheckResourceAttr(fqrn, "handler.0.secrets.secret2", "value2"),
+					resource.TestCheckResourceAttr(fqrn, "handler.0.http_headers.%", "2"),
+					resource.TestCheckResourceAttr(fqrn, "handler.0.http_headers.header-1", "value-1"),
+					resource.TestCheckResourceAttr(fqrn, "handler.0.http_headers.header-2", "value-2"),
+					resource.TestCheckResourceAttr(fqrn, "handler.0.payload", "{ \"event_type\": \"distribution/{{ .type }}\", \"client_payload\": {{ json .data}} }"),
+				),
+			},
+			{
+				ResourceName:                         fqrn,
+				ImportState:                          true,
+				ImportStateId:                        name,
+				ImportStateVerify:                    true,
+				ImportStateVerifyIdentifierAttribute: "key",
+				ImportStateVerifyIgnore:              []string{"handler.0.secrets"},
+			},
+		},
+	})
+}

--- a/pkg/artifactory/resource/webhook/resource_artifactory_webhook_release_bundle.go
+++ b/pkg/artifactory/resource/webhook/resource_artifactory_webhook_release_bundle.go
@@ -313,7 +313,7 @@ func fromReleaseBundleCriteriaAPIModel(ctx context.Context, criteriaAPIModel map
 		lo.Assign(
 			baseCriteriaAttrs,
 			map[string]attr.Value{
-				"any_release_bundle":              types.BoolValue(criteriaAPIModel["anyReleaseBundles"].(bool)),
+				"any_release_bundle":              types.BoolValue(criteriaAPIModel["anyReleaseBundle"].(bool)),
 				"registered_release_bundle_names": releaseBundleNames,
 			},
 		),

